manager/manager.py

```diff
@@ -0,0 +1,1105 @@
+
+    def _save_cfg_to_configs_file(self):
+        """
+        Save self.cfg to appdir configuration .yaml file.
+        Path objects must be converted to string (and likewise, back to Path when loaded).
+        """
+        cfg_to_save = copy.deepcopy(self.cfg)
+        self._convert_str_and_pathlib_paths(cfg_to_save, "path_to_str")
+        self._dump_configs_to_file(cfg_to_save)
+
+    def _dump_configs_to_file(self, config_dict):
+
+        with open(self._config_path, "w") as config_file:
+            yaml.dump(config_dict, config_file, sort_keys=False)
+
+    def _config_file_exists(self, prompt_on_fail: bool) -> bool:
+        """
+        Check the config file exists in the expected directory.
+
+        :param prompt_on_fail: if config file not found, warn the user.
+
+        :return: True or False
+        """
+        exists = os.path.isfile(self._config_path)
+
+        if not exists and prompt_on_fail:
+            warnings.warn(
+                "Configuration file has not been initialized. "
+                "Use make_config_file() to setup before continuing."
+            )
+
+        return exists
+
+    def _attempt_load_configs(self, prompt_on_fail: bool) -> Union[bool, dict]:
+        """
+        Attempt to load the config file. If it does not exist or crashes when attempt to load from file, return False.
+
+        :param prompt_on_fail: if config file not found, or crashes on load, warn the user.
+
+        :return: loaded dictionary, or False if not loaded.
+        """
+        if not self._config_file_exists(prompt_on_fail):
+            return False
+
+        try:
+            with open(self._config_path, "r") as config_file:
+                config_dict = yaml.full_load(config_file)
+
+            self._convert_str_and_pathlib_paths(config_dict, "str_to_path")
+
+        except:
+            config_dict = False
+
+            if prompt_on_fail:
+                self._message_user(
+                    f"Config file failed to load. Check file formatting at {self._config_path}. "
+                    f"If cannot load, re-initialise configs with make_config_file()"
+                )
+
+        return config_dict
+
+    def _convert_str_and_pathlib_paths(self, config_dict: dict, direction: str):
+        """
+        Config paths are stored as str in the .yaml but used as Path
+        in the module, so make the conversion here.
+
+        :param config_dict: self.cfg dict of configs
+        :param direction: "path_to_str" or "str_to_path"
+
+        TODO: put "local_path" and "remote_path" keys somewhere central.
+        """
+        for path_key in ["local_path", "remote_path"]:
+            if direction == "str_to_path":
+                config_dict[path_key] = Path(config_dict[path_key])
+            elif direction == "path_to_str":
+                config_dict[path_key] = config_dict[path_key].as_posix()
```

Comment:
Out of curiosity, is `as_posix()` preferred to `str()`?

Reply:
hmm good question, I used posix() with some vague notion to keep all outputs to forward slashes OS system independent, but not sure if it actually serves any purpose. 

It seems os.fspath() [is preferred](https://stackoverflow.com/questions/56477425/real-path-of-file-string-pathlib), shall default to this?

Reply:
Sounds good.

Reply:
In the end I used posix() for 'internal' functions as was running into an issue in which sections of paths generated from the windows local project were not compatible with unix system. However for public functions (e.g. project.get_local_path()) the os.fspath is used.